### PR TITLE
chipsec: 1.8.1 -> 1.10.6; unbreak

### PR DESCRIPTION
**Pull request by erdnaxe**
###### Description of changes

Update chipsec to latest version.
Also fix current chipsec kernel modules not building (https://github.com/NixOS/nixpkgs/issues/230712).

ZHF: #230712

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by erdnaxe** [comment]
Result of `nixpkgs-review pr 231806` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14_hardened.chipsec</li>
    <li>linuxKernel.packages.linux_4_14_hardened.chipsec.dist</li>
    <li>linuxKernel.packages.linux_4_19_hardened.chipsec</li>
    <li>linuxKernel.packages.linux_4_19_hardened.chipsec.dist</li>
    <li>linuxKernel.packages.linux_5_4_hardened.chipsec</li>
    <li>linuxKernel.packages.linux_5_4_hardened.chipsec.dist</li>
  </ul>
</details>
<details>
  <summary>38 packages built:</summary>
  <ul>
    <li>chipsec</li>
    <li>chipsec.dist</li>
    <li>linuxKernel.packages.linux_4_14.chipsec</li>
    <li>linuxKernel.packages.linux_4_14.chipsec.dist</li>
    <li>linuxKernel.packages.linux_4_19.chipsec</li>
    <li>linuxKernel.packages.linux_4_19.chipsec.dist</li>
    <li>linuxKernel.packages.linux_5_10.chipsec</li>
    <li>linuxKernel.packages.linux_5_10.chipsec.dist</li>
    <li>linuxKernel.packages.linux_5_10_hardened.chipsec</li>
    <li>linuxKernel.packages.linux_5_10_hardened.chipsec.dist</li>
    <li>linuxKernel.packages.linux_5_15.chipsec</li>
    <li>linuxKernel.packages.linux_5_15.chipsec.dist</li>
    <li>linuxKernel.packages.linux_5_15_hardened.chipsec</li>
    <li>linuxKernel.packages.linux_5_15_hardened.chipsec.dist</li>
    <li>linuxKernel.packages.linux_5_4.chipsec</li>
    <li>linuxKernel.packages.linux_5_4.chipsec.dist</li>
    <li>linuxKernel.packages.linux_6_1.chipsec</li>
    <li>linuxKernel.packages.linux_6_1.chipsec.dist</li>
    <li>linuxKernel.packages.linux_hardened.chipsec (linuxKernel.packages.linux_6_1_hardened.chipsec)</li>
    <li>linuxKernel.packages.linux_hardened.chipsec.dist (linuxKernel.packages.linux_6_1_hardened.chipsec.dist)</li>
    <li>linuxKernel.packages.linux_6_2.chipsec</li>
    <li>linuxKernel.packages.linux_6_2.chipsec.dist</li>
    <li>linuxKernel.packages.linux_6_3.chipsec</li>
    <li>linuxKernel.packages.linux_6_3.chipsec.dist</li>
    <li>linuxKernel.packages.linux_latest_libre.chipsec</li>
    <li>linuxKernel.packages.linux_latest_libre.chipsec.dist</li>
    <li>linuxKernel.packages.linux_libre.chipsec</li>
    <li>linuxKernel.packages.linux_libre.chipsec.dist</li>
    <li>linuxKernel.packages.linux_lqx.chipsec</li>
    <li>linuxKernel.packages.linux_lqx.chipsec.dist</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.chipsec</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.chipsec.dist</li>
    <li>linuxKernel.packages.linux_xanmod.chipsec</li>
    <li>linuxKernel.packages.linux_xanmod.chipsec.dist</li>
    <li>linuxKernel.packages.linux_xanmod_latest.chipsec (linuxKernel.packages.linux_xanmod_stable.chipsec)</li>
    <li>linuxKernel.packages.linux_xanmod_latest.chipsec.dist (linuxKernel.packages.linux_xanmod_stable.chipsec.dist)</li>
    <li>linuxKernel.packages.linux_zen.chipsec</li>
    <li>linuxKernel.packages.linux_zen.chipsec.dist</li>
  </ul>
</details>

**Comment by orangecms** [comment]
Nice, love the `if True` hack :)

I don't mind those old kernels, tbh; it works for recent ones now vs none before, so it's an improvement :)

File: pkgs/tools/security/chipsec/default.nix [hunk 1]
```diff
@@ -28,6 +29,7 @@ python3.pkgs.buildPythonApplication rec {
   nativeBuildInputs = [
     libelf
     nasm
+    pahole
```
Comment (RaitoBezarius) [inline pkgs/tools/security/chipsec/default.nix:32]:
```suggestion
] ++ kernel.moduleBuildDependencies;
```

File: pkgs/tools/security/chipsec/default.nix [hunk 2]
```diff
@@ -28,6 +29,7 @@ python3.pkgs.buildPythonApplication rec {
...
   ];
```
Comment (RaitoBezarius) [inline pkgs/tools/security/chipsec/default.nix:33]:
```suggestion
```

File: pkgs/tools/security/chipsec/default.nix [hunk 3]
```diff
@@ -4,21 +4,22 @@
 , kernel ? null
 , libelf
 , nasm
+, pahole
```
Comment (RaitoBezarius) [inline pkgs/tools/security/chipsec/default.nix:7]:
```suggestion
, kernel
```

**Review by RaitoBezarius — CHANGES_REQUESTED** [review]
Please use `kernel.moduleBuildDependencies`

File: pkgs/tools/security/chipsec/default.nix (hunk 3 above)
Comment (erdnaxe) [reply pkgs/tools/security/chipsec/default.nix]: kernel was already present, reusing using `lib.optional withDriver`.

File: pkgs/tools/security/chipsec/default.nix (hunk 1 above)
Comment (erdnaxe) [reply pkgs/tools/security/chipsec/default.nix]: Thanks! I was not aware that `kernel.moduleBuildDependencies` was preferable to `pahole`.

**Review by RaitoBezarius — APPROVED** [review]

**Comment by RaitoBezarius** [comment]
evaluation is failing

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 231806` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.hardkernel_4_14.chipsec</li>
    <li>linuxKernel.packages.hardkernel_4_14.chipsec.dist</li>
    <li>linuxPackages_hardkernel_latest.chipsec</li>
    <li>linuxPackages_hardkernel_latest.chipsec.dist</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>linuxPackages_4_14_hardened.chipsec (linuxKernel.packages.linux_4_14_hardened.chipsec)</li>
    <li>linuxPackages_4_14_hardened.chipsec.dist (linuxKernel.packages.linux_4_14_hardened.chipsec.dist)</li>
    <li>linuxPackages_4_19_hardened.chipsec (linuxKernel.packages.linux_4_19_hardened.chipsec)</li>
    <li>linuxPackages_4_19_hardened.chipsec.dist (linuxKernel.packages.linux_4_19_hardened.chipsec.dist)</li>
    <li>linuxPackages_5_4_hardened.chipsec (linuxKernel.packages.linux_5_4_hardened.chipsec)</li>
    <li>linuxPackages_5_4_hardened.chipsec.dist (linuxKernel.packages.linux_5_4_hardened.chipsec.dist)</li>
  </ul>
</details>
<details>
  <summary>38 packages built:</summary>
  <ul>
    <li>chipsec</li>
    <li>chipsec.dist</li>
    <li>linuxKernel.packages.linux_4_14.chipsec</li>
    <li>linuxKernel.packages.linux_4_14.chipsec.dist</li>
    <li>linuxKernel.packages.linux_4_19.chipsec</li>
    <li>linuxKernel.packages.linux_4_19.chipsec.dist</li>
    <li>linuxKernel.packages.linux_5_10.chipsec</li>
    <li>linuxKernel.packages.linux_5_10.chipsec.dist</li>
    <li>linuxPackages_5_10_hardened.chipsec (linuxKernel.packages.linux_5_10_hardened.chipsec)</li>
    <li>linuxPackages_5_10_hardened.chipsec.dist (linuxKernel.packages.linux_5_10_hardened.chipsec.dist)</li>
    <li>linuxKernel.packages.linux_5_15.chipsec</li>
    <li>linuxKernel.packages.linux_5_15.chipsec.dist</li>
    <li>linuxPackages_5_15_hardened.chipsec (linuxKernel.packages.linux_5_15_hardened.chipsec)</li>
    <li>linuxPackages_5_15_hardened.chipsec.dist (linuxKernel.packages.linux_5_15_hardened.chipsec.dist)</li>
    <li>linuxKernel.packages.linux_5_4.chipsec</li>
    <li>linuxKernel.packages.linux_5_4.chipsec.dist</li>
    <li>linuxPackages.chipsec (linuxKernel.packages.linux_6_1.chipsec)</li>
    <li>linuxPackages.chipsec.dist (linuxKernel.packages.linux_6_1.chipsec.dist)</li>
    <li>linuxPackages_hardened.chipsec (linuxPackages_6_1_hardened.chipsec)</li>
    <li>linuxPackages_hardened.chipsec.dist (linuxPackages_6_1_hardened.chipsec.dist)</li>
    <li>linuxKernel.packages.linux_6_2.chipsec</li>
    <li>linuxKernel.packages.linux_6_2.chipsec.dist</li>
    <li>linuxPackages_latest.chipsec (linuxKernel.packages.linux_6_3.chipsec)</li>
    <li>linuxPackages_latest.chipsec.dist (linuxKernel.packages.linux_6_3.chipsec.dist)</li>
    <li>linuxPackages_latest-libre.chipsec (linuxKernel.packages.linux_latest_libre.chipsec)</li>
    <li>linuxPackages_latest-libre.chipsec.dist (linuxKernel.packages.linux_latest_libre.chipsec.dist)</li>
    <li>linuxPackages-libre.chipsec (linuxKernel.packages.linux_libre.chipsec)</li>
    <li>linuxPackages-libre.chipsec.dist (linuxKernel.packages.linux_libre.chipsec.dist)</li>
    <li>linuxPackages_lqx.chipsec (linuxKernel.packages.linux_lqx.chipsec)</li>
    <li>linuxPackages_lqx.chipsec.dist (linuxKernel.packages.linux_lqx.chipsec.dist)</li>
    <li>linuxPackages_testing_bcachefs.chipsec (linuxKernel.packages.linux_testing_bcachefs.chipsec)</li>
    <li>linuxPackages_testing_bcachefs.chipsec.dist (linuxKernel.packages.linux_testing_bcachefs.chipsec.dist)</li>
    <li>linuxPackages_xanmod.chipsec (linuxKernel.packages.linux_xanmod.chipsec)</li>
    <li>linuxPackages_xanmod.chipsec.dist (linuxKernel.packages.linux_xanmod.chipsec.dist)</li>
    <li>linuxPackages_xanmod_latest.chipsec (linuxKernel.packages.linux_xanmod_latest.chipsec ,linuxPackages_xanmod_stable.chipsec)</li>
    <li>linuxPackages_xanmod_latest.chipsec.dist (linuxKernel.packages.linux_xanmod_latest.chipsec.dist ,linuxPackages_xanmod_stable.chipsec.dist)</li>
    <li>linuxPackages_zen.chipsec (linuxKernel.packages.linux_zen.chipsec)</li>
    <li>linuxPackages_zen.chipsec.dist (linuxKernel.packages.linux_zen.chipsec.dist)</li>
  </ul>
</details>

**Comment by RaitoBezarius** [comment]
Anyway, it sounds like already very good.

**Comment by RaitoBezarius** [comment]
We can do the rest in a follow-up.

**Pull request merged**